File: WhatsAppAutoReply/app/src/main/java/com/whatsappauto/reply/MainActivity.java
package com.whatsappauto.reply;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;
import com.whatsappauto.reply.databinding.ActivityMainBinding;
import com.whatsappauto.reply.fragments.AutoReplyFragment;
import com.whatsappauto.reply.fragments.ScheduledMessageFragment;
import com.whatsappauto.reply.services.MessageSchedulerService;
import com.whatsappauto.reply.utils.WhatsAppHelper;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    private int currentTab = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setSupportActionBar(binding.toolbar);
        setupViewPager();
        setupFab();

        // Start the message scheduler service
        startService(new Intent(this, MessageSchedulerService.class));

        // Check if accessibility service is enabled
        checkAccessibilityService();
    }

    private void setupViewPager() {
        ViewPagerAdapter adapter = new ViewPagerAdapter(this);
        binding.viewPager.setAdapter(adapter);

        new TabLayoutMediator(binding.tabs, binding.viewPager,
                (tab, position) -> {
                    switch (position) {
                        case 0:
                            tab.setText("Auto Reply");
                            break;
                        case 1:
                            tab.setText("Scheduled Messages");
                            break;
                    }
                }
        ).attach();

        binding.tabs.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                currentTab = tab.getPosition();
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {}

            @Override
            public void onTabReselected(TabLayout.Tab tab) {}
        });
    }

    private void setupFab() {
        binding.fab.setOnClickListener(view -> {
            Fragment currentFragment = getSupportFragmentManager()
                    .findFragmentByTag("f" + currentTab);
            
            if (currentFragment instanceof AutoReplyFragment) {
                ((AutoReplyFragment) currentFragment).showAddDialog();
            } else if (currentFragment instanceof ScheduledMessageFragment) {
                ((ScheduledMessageFragment) currentFragment).showAddDialog();
            }
        });
    }

    private void checkAccessibilityService() {
        if (!WhatsAppHelper.isAccessibilityServiceEnabled(this)) {
            Toast.makeText(this, "Please enable accessibility service", Toast.LENGTH_LONG).show();
            WhatsAppHelper.openAccessibilitySettings(this);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.action_settings) {
            WhatsAppHelper.openAccessibilitySettings(this);
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private class ViewPagerAdapter extends FragmentStateAdapter {
        public ViewPagerAdapter(FragmentActivity fa) {
            super(fa);
        }

        @Override
        public Fragment createFragment(int position) {
            switch (position) {
                case 0:
                    return new AutoReplyFragment();
                case 1:
                    return new ScheduledMessageFragment();
                default:
                    return new AutoReplyFragment();
            }
        }

        @Override
        public int getItemCount() {
            return 2;
        }
    }
}

File: app/src/main/java/com/whatsappautoresponder/MainActivity.kt
package com.whatsappautoresponder

import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.google.android.material.tabs.TabLayoutMediator
import com.whatsappautoresponder.databinding.ActivityMainBinding
import com.whatsappautoresponder.ui.AutoReplyRulesFragment
import com.whatsappautoresponder.ui.ScheduledMessagesFragment
import com.whatsappautoresponder.utils.AccessibilityUtils
import com.whatsappautoresponder.utils.SharedPreferencesManager

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var prefsManager: SharedPreferencesManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        prefsManager = SharedPreferencesManager(this)
        
        setupToolbar()
        setupViewPager()
        setupFab()
        updateServiceStatus()
        setupAutoReplySwitch()
    }

    override fun onResume() {
        super.onResume()
        updateServiceStatus()
    }

    private fun setupToolbar() {
        setSupportActionBar(binding.toolbar)
        supportActionBar?.title = getString(R.string.app_name)
    }

    private fun setupViewPager() {
        val adapter = ViewPagerAdapter(this)
        binding.viewPager.adapter = adapter

        TabLayoutMediator(binding.tabLayout, binding.viewPager) { tab, position ->
            tab.text = when (position) {
                0 -> getString(R.string.auto_reply_rules)
                1 -> getString(R.string.scheduled_messages)
                else -> ""
            }
        }.attach()
    }

    private fun setupFab() {
        binding.fabAdd.setOnClickListener {
            when (binding.viewPager.currentItem) {
                0 -> {
                    // Add auto reply rule
                    val intent = Intent(this, AddAutoReplyRuleActivity::class.java)
                    startActivity(intent)
                }
                1 -> {
                    // Add scheduled message
                    val intent = Intent(this, AddScheduledMessageActivity::class.java)
                    startActivity(intent)
                }
            }
        }
    }

    private fun updateServiceStatus() {
        val isAccessibilityEnabled = AccessibilityUtils.isAccessibilityServiceEnabled(this)
        
        if (isAccessibilityEnabled) {
            binding.tvServiceStatus.text = getString(R.string.auto_reply_enabled)
            binding.tvServiceStatus.setTextColor(getColor(R.color.whatsapp_green))
            binding.switchAutoReply.isEnabled = true
        } else {
            binding.tvServiceStatus.text = getString(R.string.accessibility_not_enabled)
            binding.tvServiceStatus.setTextColor(getColor(R.color.error))
            binding.switchAutoReply.isEnabled = false
            binding.switchAutoReply.isChecked = false
        }
    }

    private fun setupAutoReplySwitch() {
        binding.switchAutoReply.isChecked = prefsManager.isAutoReplyEnabled()
        
        binding.switchAutoReply.setOnCheckedChangeListener { _, isChecked ->
            prefsManager.setAutoReplyEnabled(isChecked)
        }
        
        binding.tvServiceStatus.setOnClickListener {
            if (!AccessibilityUtils.isAccessibilityServiceEnabled(this)) {
                openAccessibilitySettings()
            }
        }
    }

    private fun openAccessibilitySettings() {
        val intent = Intent(Settings.ACTION_ACCESSIBILITY_SETTINGS)
        startActivity(intent)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_settings -> {
                val intent = Intent(this, SettingsActivity::class.java)
                startActivity(intent)
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    private class ViewPagerAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {
        override fun getItemCount(): Int = 2

        override fun createFragment(position: Int): Fragment {
            return when (position) {
                0 -> AutoReplyRulesFragment()
                1 -> ScheduledMessagesFragment()
                else -> throw IllegalArgumentException("Invalid position: $position")
            }
        }
    }
}